Description: adjust ejabberd.yml.example to sane defaults
 ejabberd's default configuration needs to be altered to be
 useful on Debian systems
Author: Philipp Huebner <debalance@debian.org>

Index: ejabberd/ejabberd.yml.example
===================================================================
--- ejabberd.orig/ejabberd.yml.example
+++ ejabberd/ejabberd.yml.example
@@ -29,16 +29,31 @@
 hosts:
   - "localhost"
 
+## rotation: Disable ejabberd's internal log rotation, as the Debian package
+## uses logrotate(8).
 loglevel: 4
-log_rotate_size: 10485760
-log_rotate_date: ""
-log_rotate_count: 1
-log_rate_limit: 100
+log_rotate_count: 0
 
 certfiles:
   - "/etc/letsencrypt/live/localhost/fullchain.pem"
   - "/etc/letsencrypt/live/localhost/privkey.pem"
 
+## TLS configuration
+define_macro:
+  'TLS_CIPHERS': "HIGH:!aNULL:!eNULL:!3DES:@STRENGTH"
+  'TLS_OPTIONS':
+    - "no_sslv3"
+    - "cipher_server_preference"
+    - "no_compression"
+  ## 'DH_FILE': "/path/to/dhparams.pem" # generated with: openssl dhparam -out dhparams.pem 2048
+
+c2s_ciphers: 'TLS_CIPHERS'
+s2s_ciphers: 'TLS_CIPHERS'
+c2s_protocol_options: 'TLS_OPTIONS'
+s2s_protocol_options: 'TLS_OPTIONS'
+## c2s_dhfile: 'DH_FILE'
+## s2s_dhfile: 'DH_FILE'
+
 listen:
   -
     port: 5222
@@ -48,6 +63,7 @@
     shaper: c2s_shaper
     access: c2s
     starttls_required: true
+    protocol_options: 'TLS_OPTIONS'
   -
     port: 5269
     ip: "::"
@@ -65,13 +81,26 @@
     web_admin: true
     captcha: true
     tls: true
+    protocol_options: 'TLS_OPTIONS'
   -
     port: 5280
     ip: "::"
     module: ejabberd_http
     web_admin: true
 
-s2s_use_starttls: optional
+## Disabling digest-md5 SASL authentication. digest-md5 requires plain-text
+## password storage (see auth_password_format option).
+disable_sasl_mechanisms: 
+  - "digest-md5"
+  - "X-OAUTH2"
+
+s2s_use_starttls: required
+
+## Store the plain passwords or hashed for SCRAM:
+auth_password_format: scram
+
+## Full path to a script that generates the image.
+## captcha_cmd: "/usr/share/ejabberd/captcha.sh
 
 acl:
   local:
@@ -110,9 +139,9 @@
   "admin access":
     who:
       - access:
-          - allow:
-            - acl: loopback
-            - acl: admin
+        - allow:
+          - acl: loopback
+          - acl: admin
       - oauth:
         - scope: "ejabberd:admin"
         - access:
@@ -156,12 +185,16 @@
   mod_carboncopy: {}
   mod_client_state: {}
   mod_configure: {}
+  ## mod_delegation: {}   # for xep0356
   mod_disco: {}
   mod_fail2ban: {}
   mod_http_api: {}
   mod_http_upload:
     put_url: "https://@HOST@:5443/upload"
   mod_last: {}
+  mod_pres_counter:
+    count: 5
+    interval: 60
   mod_mam:
     ## Mnesia is limited to 2GB, better to use an SQL backend
     ## For small servers SQLite is a good fit and is very easy
@@ -196,7 +229,7 @@
       ## Change from "whitelist" to "open" to enable OMEMO support
       ## See https://github.com/processone/ejabberd/issues/2425
       "eu.siacs.conversations.axolotl.*":
-        access_model: whitelist
+        access_model: open
       ## Avoid buggy clients to make their bookmarks public
       "storage:bookmarks":
         access_model: whitelist
@@ -213,6 +246,7 @@
     versioning: true
   mod_s2s_dialback: {}
   mod_shared_roster: {}
+  mod_sic: {}
   mod_stream_mgmt:
     resend_on_timeout: if_offline
   mod_vcard: {}
